src/main/resources/assets/proton/blockstates/thatch_stairs.json

```diff
@@ -0,0 +1,44 @@
+    "facing=west,half=bottom,shape=inner_left":  { "model": "proton:block/thatch_stairs_inner", "y": 90, "uvlock": true },
+    "facing=south,half=bottom,shape=inner_left": { "model": "proton:block/thatch_stairs_inner" },
+    "facing=north,half=bottom,shape=inner_left": { "model": "proton:block/thatch_stairs_inner", "y": 180, "uvlock": true },
+    "facing=east,half=top,shape=straight":  { "model": "proton:block/thatch_stairs", "x": 180, "uvlock": true },
+    "facing=west,half=top,shape=straight":  { "model": "proton:block/thatch_stairs", "x": 180, "y": 180, "uvlock": true },
+    "facing=south,half=top,shape=straight": { "model": "proton:block/thatch_stairs", "x": 180, "y": 90, "uvlock": true },
+    "facing=north,half=top,shape=straight": { "model": "proton:block/thatch_stairs", "x": 180, "y": 270, "uvlock": true },
+    "facing=east,half=top,shape=outer_right":  { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 90, "uvlock": true },
+    "facing=west,half=top,shape=outer_right":  { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 270, "uvlock": true },
+    "facing=south,half=top,shape=outer_right": { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 180, "uvlock": true },
+    "facing=north,half=top,shape=outer_right": { "model": "proton:block/thatch_stairs_outer", "x": 180, "uvlock": true },
+    "facing=east,half=top,shape=outer_left":  { "model": "proton:block/thatch_stairs_outer", "x": 180, "uvlock": true },
+    "facing=west,half=top,shape=outer_left":  { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 180, "uvlock": true },
+    "facing=south,half=top,shape=outer_left": { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 90, "uvlock": true },
+    "facing=north,half=top,shape=outer_left": { "model": "proton:block/thatch_stairs_outer", "x": 180, "y": 270, "uvlock": true },
+    "facing=east,half=top,shape=inner_right":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 90, "uvlock": true },
+    "facing=west,half=top,shape=inner_right":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 270, "uvlock": true },
+    "facing=south,half=top,shape=inner_right": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 180, "uvlock": true },
+    "facing=north,half=top,shape=inner_right": { "model": "proton:block/thatch_stairs_inner", "x": 180, "uvlock": true },
+    "facing=east,half=top,shape=inner_left":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "uvlock": true },
+    "facing=west,half=top,shape=inner_left":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 180, "uvlock": true },
+    "facing=south,half=top,shape=inner_left": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 90, "uvlock": true },
+    "facing=north,half=top,shape=inner_left": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 270, "uvlock": true }
+  }
+}
```

Comment:
I don't know much about blockstate variants and such, I'll have to trust you that you did it right